src/MainWindow.vala

```diff
@@ -427,6 +461,69 @@ namespace PantheonCalculator {
             entry.set_position (position - 1);
         }
 
+        private void button_memory_store_clicked () {
+            if (entry.get_text () != "") {
+                try {
+                    /* This is necessary because we don't want to accept non-numeric values.
+                     * Moreover, it allows the saving in memory of the result of complex mathematical
+                     * functions by calculating the result separately and allowing the user to recall
+                     * it using the MR button (e.g. "√4" will store the value "2,0" in the variable).
+                     */
+                    var output = eval.evaluate (entry.get_text (), decimal_places);
+                    memory_value = double.parse (output);
+
+                    button_mr.set_sensitive (true);
+                    button_mc.set_sensitive (true);
+                } catch (Core.OUT_ERROR e) {
+                    infobar_label.label = e.message;
+                    infobar.revealed = true;
+                }
+            }
+        }
+
+        // Addition and subtraction to and from memory value are done silently,
+        // updating the value without informing the user, as in real-life calculators.
+        private void button_memory_add_clicked () {
+            if (entry.get_text () != "") {
+                try {
+                    var output = eval.evaluate (entry.get_text (), decimal_places);
+                    memory_value += double.parse (output);
+                } catch (Core.OUT_ERROR e) {
+                    infobar_label.label = e.message;
+                    infobar.revealed = true;
+                }
+            }
+        }
+
+        private void button_memory_subtract_clicked () {
+            if (entry.get_text () != "") {
+                try {
+                    var output = eval.evaluate (entry.get_text (), decimal_places);
+                    memory_value -= double.parse (output);
+                } catch (Core.OUT_ERROR e) {
+                    infobar_label.label = e.message;
+                    infobar.revealed = true;
+                }
+            }
+        }
+
+        private void button_memory_clear_clicked () {
+            memory_value = 0;
+            button_mr.sensitive = false;
+            button_mc.sensitive = false;
+        }
+
+        private void button_gt_clicked () {
+            action_clear ();
+
+            double grand_total = 0;
+            history.foreach ((list_entry) => {
+                grand_total += double.parse (list_entry.output);
+            });
+
+            entry.set_text (grand_total.to_string ());
```

Comment:
For some reason, the entry cursor ends up on the wrong side (left) of grand total.